### ISSUE TITLE
Improve "Add To" login modal

### ISSUE THREAD

**Issue by kmthorn** [OregonDigital/OD2#2077]
### Descriptive summary
As a user when I try to add a work to my collection and I am not logged in then I would like an actionable modal.

### Expected behavior
- When not logged in and attempting to add an item to my collection the modal should include login actions. 
- Users should be returned to where they tried logging in from
- Developer may choose if the modal contains the same login actions as the login button or a simple link to the login page. Whatever is easiest, but a user must be brought back to where they started.
  - NOTE: If we go with using the login actions from the login modal we'll be deviating from the wireframes, this is ok as long as we stick to the exact same design as the login modal

### Related work
#2076 

### Accessibility Concerns
- Compare modal to existing login modal

**Comment by CGillen** [comment]
Unless we think we want something other than squishing the same login modal underneath a message that says `Please login to access your collections` we shouldn't need a wireframe.
I'll make a strong note on here that we might deviate from the wireframe if the modal login version is possible

**Comment by shieldsb** [comment]
QA fail.  After logging in, I'm not returned to where point I prompted to login.  After logging in, it goes to the homepage

**Comment by shieldsb** [comment]
QA fail. After logging in, I'm not returned to where point I prompted to login. After logging in, it goes to the homepage

**Comment by shieldsb** [comment]
QA fail.  After I logged in, I returned to a thumbnail for a different work. (Also confirmed by trying to add a different work, and landed at the same thumbnail as the first test- see image below). 

What I tried to add to my collection before logging in: 
![image](https://user-images.githubusercontent.com/53542788/196282857-6e1f5f9b-a153-49c7-aedc-292ccbe5860a.png)

Where I ended up after log in: 
![image](https://user-images.githubusercontent.com/53542788/196282625-f52a97c3-0ae0-4ec8-91d9-41a4bea72d7b.png)
